scripts/bootstrap.sh.in

```diff
@@ -144,28 +144,42 @@ retry_dns_count: 3
 EOF
 }
 
-set_salt_command() {
-    echo "Setting salt master command"
-    # shellcheck disable=SC2046
-    SALT_MASTER_CALL="crictl exec -i $(crictl ps -q --label io.kubernetes.container.name=salt-master --label io.kubernetes.pod.namespace=kube-system)"
-}
+get_salt_container() {
+    local -r max_retries=${1:-10}
+    local salt_container='' attempts=0
 
-sync_salt() {
-    $SALT_MASTER_CALL salt '*' saltutil.sync_all refresh=True \
-        saltenv=metalk8s-@@VERSION
-    $SALT_MASTER_CALL salt '*' mine.update
-    local SALT_CONTAINER
-    SALT_CONTAINER="$(crictl ps -q \
+    while [ -z "$salt_container" ] && [ $attempts -lt $max_retries ]; do
+        salt_container="$(crictl ps -q \
             --label io.kubernetes.pod.namespace=kube-system \
             --label io.kubernetes.container.name=salt-master \
             --state Running)"
-    test -n "$SALT_CONTAINER"
-    crictl exec -i "$SALT_CONTAINER" \
-        salt-run saltutil.sync_all saltenv=metalk8s-@@VERSION
-    crictl exec -i "$SALT_CONTAINER" \
-        salt-run saltutil.sync_roster saltenv=metalk8s-@@VERSION
-    crictl exec -i "$SALT_CONTAINER" \
-        salt-run metalk8s_saltutil.sync_auth saltenv=metalk8s-@@VERSION
+        (( attempts++ ))
+    done
+
+    if [ -z "$salt_container" ]; then
+        echo "Failed to find a running 'salt-master' container" >&2
+        exit 1
+    fi
+
+    echo "$salt_container"
+}
+
+sync_salt_master() {
+    local -r salt_master_call="crictl exec -i $(get_salt_container)"
+
+    $salt_master_call salt-run saltutil.sync_all saltenv=metalk8s-@@VERSION
+    $salt_master_call salt-run saltutil.sync_roster saltenv=metalk8s-@@VERSION
+    $salt_master_call salt-run metalk8s_saltutil.sync_auth \
+        saltenv=metalk8s-@@VERSION
+}
+
+sync_salt_minion() {
+    local -r target=${1:-'*'}
+    local -r salt_master_call="crictl exec -i $(get_salt_container)"
+
+    $salt_master_call salt "$target" saltutil.sync_all refresh=True \
```

Comment:
All interactions in this file targetting a minion will target the bootstrap minion, so can't you use `salt-call` on the host instead?

Reply:
Indeed, that's simpler. Will do!